braid/_braid.h

```diff
@@ -75,13 +75,13 @@ void _braid_ErrorHandler(const char *filename, braid_Int line, braid_Int ierr, c
  * Re-allocation macro 
  **/
 #define _braid_TReAlloc(ptr, type, count) \
-( (type *)realloc((char *)ptr, (size_t)(sizeof(type) * (count))) )
+( (type *)realloc((braid_Byte *)ptr, (size_t)(sizeof(type) * (count))) )
```

Comment:
I still don't see the reason for `braid_Byte`.  Will this ever be anything other than `char`?  Is this useful to users?

Reply:
@cadaverous-lives What did you and Rob decide about braid_Byte?

Reply:
It's coincidence that all of the other types have `braid_` names (it wasn't for style reasons).  The real reason for this is that we may need to change the underlying type to something different.  We need to do this in hypre to get around the fact that `int` is sometimes too small (for large problems, we need to deal with numbers bigger than 2 billion) or because we want to do arithmetic in a different precision.  We may need to do this in braid for similar reasons.

There is no reason for `braid_Byte` to ever be anything different than `char`.  However, I guess it doesn't hurt to leave it as is either, so let's do that.

Reply:
I get that. Jacob and I were discussing this today and this is what we were thinking as well. I can always remove it after the pull request!